src/rebar_app_discover.erl

```diff
@@ -154,28 +153,35 @@ parse_profile_deps(Profile, Name, Deps, Opts, State) ->
 
 %% @doc Find the app-level config and return the state updated
 %% with the relevant app-level data.
--spec project_app_config(rebar_app_info:t(), rebar_state:t()) ->
+-spec project_app_config(rebar_app_info:t(), rebar_dict(), rebar_state:t()) ->
     {Config, rebar_state:t()} when
       Config :: [any()].
-project_app_config(AppInfo, State) ->
+project_app_config(AppInfo, Default, State) ->
     C = rebar_config:consult(rebar_app_info:dir(AppInfo)),
+    AppInfo1 = rebar_app_info:update_opts(AppInfo, Default, C),
+    {AppInfo2, State1} = maybe_reset_hooks_plugins(AppInfo1, State),
+    {AppInfo2, State1}.
+
+-spec maybe_reset_hooks_plugins(AppInfo, State) ->  {AppInfo, State} when
+      AppInfo :: rebar_app_info:t(),
+      State :: rebar_state:t().
+maybe_reset_hooks_plugins(AppInfo, State) ->
     Dir = rebar_app_info:dir(AppInfo),
-    Opts = maybe_reset_hooks(Dir, rebar_state:opts(State), State),
-    {C, rebar_state:opts(State, Opts)}.
-
-%% @private Check if the app is at the root of the project.
-%% If it is, then drop the hooks from the config so they aren't run twice
--spec maybe_reset_hooks(file:filename(), Opts, rebar_state:t()) -> Opts when
-      Opts :: rebar_dict().
-maybe_reset_hooks(Dir, Opts, State) ->
     case ec_file:real_dir_path(rebar_dir:root_dir(State)) of
         Dir ->
             CurrentProfiles = rebar_state:current_profiles(State),
-            reset_hooks(Opts, CurrentProfiles);
+            Opts = reset_hooks(rebar_state:opts(State), CurrentProfiles),
+            State1 = rebar_state:opts(State, Opts),
+
+            %% set plugins to empty since this is an app at the top level
```

Comment:
I guess the implied tail of this comment is "since they have been handled in rebar3:run_aux" ?

Reply:
Yea, I can add that.